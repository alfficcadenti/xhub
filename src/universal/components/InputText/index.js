import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {FormTextArea} from '@homeaway/react-form-components';

class InputText extends Component {
    constructor(props) {
        super(props);
        this.state = {
            value: '',
            error: ''
        };
    }

    onChange = (event) => {
        this.setState({
            value: event.target.value
        });
    }

    render() {
        const {
            id,
            question,
            value
        } = this.props;

        return (
            <FormTextArea
                key={id}
                id={id}
                name={id}
                label={question}
                onChange={this.onChange}
                value={this.state.value || value}
                autoExpand
            />
        );
    }
}

InputText.defaultProps = {
    id: null,
    question: ''
};

InputText.propTypes = {
    id: PropTypes.string,
    question: PropTypes.string,
};

export default InputText;