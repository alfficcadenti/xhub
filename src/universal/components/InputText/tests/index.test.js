import React from 'react';
import {mount} from 'enzyme';
import chai, {expect} from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';
import InputText from '../index';
import renderer from 'react-test-renderer';
chai.use(chaiJestSnapshot);

const question = [
    {id: 0, question: '# Availability Zones Deployed to'}
]

describe('<InputText /> ', () => {

    beforeEach(function() {
        chaiJestSnapshot.setFilename(__filename + ".snap");
      });

    it('renders correctly', () => {
        const props = question
        const wrapper = mount(<InputText {...props}/>);
        expect(wrapper).to.have.lengthOf(1);
    });

    it('matches the snapshot', () => {
        const props = question;
        chaiJestSnapshot.setTestName("matches the snapshot");
        const wrapper = renderer.create(<InputText {...props}/>);
        expect(wrapper).to.matchSnapshot();
    });

    describe('onChange()', () => {
        it('changes the state value when user input text', () => {
            const props = question
            const wrapper = mount(<InputText {...props}/>);
            wrapper.find('textarea').simulate('change', {
                target: { value: 'test' }
              })
            expect(wrapper.state('value')).to.be.eql('test');
        });
    })
})