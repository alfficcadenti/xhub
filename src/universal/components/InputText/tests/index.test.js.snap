// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="form-group floating-label empty"
>
  <textarea
    aria-label="# Availability Zones Deployed to"
    className="FormTextArea__textarea form-control FormTextArea__textarea--auto-expand"
    id="0"
    name="0"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="# Availability Zones Deployed to"
    value={undefined}
  />
  <label
    className="FormTextArea__floating-label"
    htmlFor="0"
  >
    # Availability Zones Deployed to
  </label>
</div>
`;
